django/db/models/base.py

```diff
@@ -91,7 +91,7 @@ def __new__(cls, name, bases, attrs, **kwargs):
         # method to type.__new__() so that they're properly initialized
         # (i.e. __set_name__()).
         contributable_attrs = {}
-        for obj_name, obj in list(attrs.items()):
```

Comment:
It's done on purpose. We modify `attrs` in this loop so `list()` is used to create a copy.

Reply:
Pleas check attantively.

This loop contains **just four lines**, and there modifies just two vars:
**contributable_attrs** and **new_attrs**

Not [one](https://github.com/django/django/blob/a59de6e89e8dc1f3e71c9a5a5bbceb373ea5247e/django/db/models/base.py#L93) nor [another](https://github.com/django/django/blob/a59de6e89e8dc1f3e71c9a5a5bbceb373ea5247e/django/db/models/base.py#L85) is **attrs**.

I suppose this code was copy-pasted from another place like [this](https://github.com/django/django/blob/a59de6e89e8dc1f3e71c9a5a5bbceb373ea5247e/django/forms/forms.py#L27).

Reply:
True, it's unnecessary since 58ad030d05fa50cfed327368ab61defca3303e02.

It's not a copy&paste code in was done on purpose in the original patch, see a68ea231012.